x-pack/plugins/security_solution/server/lib/detection_engine/signals/build_bulk_body.ts

```diff
@@ -137,14 +162,31 @@ export const buildSignalFromEvent = (
   event: BaseSignalHit,
   ruleSO: SavedObject<AlertAttributes>,
   applyOverrides: boolean,
-  mergeStrategy: ConfigType['alertMergeStrategy']
+  mergeStrategy: ConfigType['alertMergeStrategy'],
+  buildReasonMessage: BuildReasonMessage
 ): SignalHit => {
   const mergedEvent = getMergeStrategy(mergeStrategy)({ doc: event });
   const rule = applyOverrides
     ? buildRuleWithOverrides(ruleSO, mergedEvent._source ?? {})
     : buildRuleWithoutOverrides(ruleSO);
+  const timestamp = new Date().toISOString();
+  let hostName = '';
+  let userName = '';
+  if (mergedEvent.fields) {
+    hostName = mergedEvent.fields['host.name'] != null ? mergedEvent.fields['host.name'] : hostName;
+    userName = mergedEvent.fields['user.name'] != null ? mergedEvent.fields['user.name'] : userName;
+  }
+  const reason = buildReasonMessage({
+    alertName: rule.name,
+    alertRiskScore: ruleSO.attributes.params.riskScore,
```

Comment:
same as above

Reply:
Nice catch on all of these. Thanks for pointing each of them out 💪🏾